### PR TITLE
fix: #89290 CSI Inline volume panic when calling applyFSGroup

### PR DESCRIPTION
**Pull request by lujinda**
#89290 

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fix: CSI Inline volume panic when calling applyFSGroup 
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/volume/csi/csi_mounter.go [hunk 1]
```diff
@@ -383,15 +383,16 @@ func (c *csiMountMgr) applyFSGroup(fsType string, fsGroup *int64, fsGroupChangeP
...
-
-		accessModes := c.spec.PersistentVolume.Spec.AccessModes
-		if c.spec.PersistentVolume.Spec.AccessModes == nil {
-			klog.V(4).Info(log("mounter.SetupAt WARNING: skipping fsGroup, access modes not provided"))
-			return nil
-		}
-		if !hasReadWriteOnce(accessModes) {
-			klog.V(4).Info(log("mounter.SetupAt WARNING: skipping fsGroup, only support ReadWriteOnce access mode"))
-			return nil
+		if c.volumeLifecycleMode == storage.VolumeLifecyclePersistent {
```
Comment (msau42) [inline pkg/volume/csi/csi_mounter.go]: Can we add a unit test for this function?

**Review by msau42 — COMMENTED** [review]
This will call ChangeVolumeOwnership on any ephemeral volume type that sets fsType. I think this is an ok assumption to make, but let's call it out in the release notes.

/assign @pohly @gnufied 
/ok-to-test

**Comment by gnufied** [comment]
@msau42 it is about time we add `fsGroup` capability in CSIDriver object I think.

**Comment by lujinda** [comment]
> @msau42 it is about time we add `fsGroup` capability in CSIDriver object I think.

Why? Maybe this is not necessary

**Comment by gnufied** [comment]
@lujinda because a inline volume without fstype might require permission change too. can you sign ccla btw?

**Comment by lujinda** [comment]
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.

signed

**Comment by lujinda** [comment]
> @lujinda because a inline volume without fstype might require permission change too. can you sign ccla btw?

signed. Thanks for reminding

**Comment by gnufied** [comment]
@lujinda can you add unit tests that @msau42 asked in above comment before we can merge this?

File: pkg/volume/csi/csi_mounter.go (hunk 1 above)
Comment (lujinda) [reply pkg/volume/csi/csi_mounter.go]: added

**Comment by lujinda** [comment]
> @lujinda can you add unit tests that @msau42 asked in above comment before we can merge this?

added

**Comment by pohly** [comment]
> This will call ChangeVolumeOwnership on any ephemeral volume type that sets fsType. I think this is an ok assumption to make, but let's call it out in the release notes.

So in other words, it will *not* be called when fsType is not set? Why is that limitation necessary?

To me it seems reasonable to combine an inline volume using the driver's default filesystem with ownership change. A driver might not support different filesystems at all and (if carefully written) might reject unknown values.

**Comment by gnufied** [comment]
@pohly I tend to agree but traditionally the idea was - a CSI volume without fstype could be a shared filesystem (like `nfs`) and hence may not need `fsGroup` change but with inline/ephemeral CSI volumes that assumption is not valid. TBH - we always knew that heuristics we use here is fragile but IMO we should address them in a separate PR.

**Comment by gnufied** [comment]
@lujinda can you add a release-note and also fix your commit message so as k8s-bot is happy?

**Comment by msau42** [comment]
@lujinda will you be able to rebase and fix the commit message?

**Comment by lujinda** [comment]
> @lujinda: The following tests **failed**, say `/retest` to rerun all failed tests:
> 
> Test name	Commit	Details	Rerun command
> pull-kubernetes-node-e2e-containerd	[c718953](https://github.com/kubernetes/kubernetes/commit/c718953f2c6c50c8c98f6d76027a8c6a64359d40)	[link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/89291/pull-kubernetes-node-e2e-containerd/1246354797431361536/)	`/test pull-kubernetes-node-e2e-containerd`
> pull-kubernetes-bazel-test	[573f620](https://github.com/kubernetes/kubernetes/commit/573f620ff0dbf9024538cd346bee0011376db00c)	[link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/89291/pull-kubernetes-bazel-test/1301896040219152387)	`/test pull-kubernetes-bazel-test`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=89291). [Your PR dashboard](https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3lujinda). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open) when you hit one in your PR.

/retest

**Pull request closed without merging**